src/twisted/internet/defer.py

```diff
@@ -2116,14 +2116,13 @@ def _addCancelCallbackToDeferred(
     @param status: a L{_CancellationStatus} tracking the current status of C{gen}
     """
     it.callbacks, tmp = [], it.callbacks
-    it.addErrback(_handleCancelInlineCallbacks, status)
+    it = it.addErrback(_handleCancelInlineCallbacks, status)
```

Comment:
not sure what is going on here.

Reply:
It now makes sense. I have no idea why I have not observed it... Friday night, 11PM :(